presto-docs/src/main/sphinx/spill/overview.rst

```diff
@@ -0,0 +1,86 @@
+or per node limits.
+
+The mechanism is similar to OS level page swapping. However, it is
+implemented on the application level to address specific needs of Presto.
+
+Properties related to spilling are described in :ref:`tuning-spilling`.
+
+Memory management and spill
+---------------------------
+
+By default, Presto kills queries if the memory requested by the query execution
+exceeds session properties ``query_max_memory`` or
+``query_max_memory_per_node``. This mechanism ensures fairness in allocation
+of memory to queries and prevents deadlock caused by memory allocation.
+It is efficient when there is a lot of small queries in the cluster, but
+leads to killing large queries that don't stay within the limits.
+
+To overcome this inefficiency, the concept of revocable memory was introduced. A
+query can request memory that does not count toward the limits, but this memory
+can be revoked by the memory manager at any time. When memory is revoked, the
+query runner spills intermediate data from memory to disk and continues to
+process it later.
+
+In practice, when the cluster is idle, and all memory is available, a memory
+intensive query may use all of the memory in the cluster. On the other hand,
+when the cluster does not have much free memory, the same query may be forced to
+use disk as storage for intermediate data. A query that is forced to spill to
+disk may have a longer execution time by orders of magnitude than a query that
+runs completely in memory.
+
+Please note that enabling spill-to-disk does not guarantee execution of all
+memory intensive queries. It is still possible that the query runner will fail
+to divide intermediate data into chunks small enough that every chunk fits into
+memory, leading to `Out of memory` errors while loading the data from disk.
+
+Supported operations
+------------------------
+
+Not all operations support spilling to disk, and each handles spilling
+differently. Currently, the mechanism is implemented for the following
+operations.
+
+Joins
+^^^^^
+
+During the join operation, one of the tables being joined is stored in memory.
+This table is called the build table. The rows from the other table stream
+through and are passed onto the next operation if they match rows in the build
+table. The most memory-intensive part of the join is this build table.
+
+When the task concurrency is greater than one, the build table is partitioned.
+The number of partitions is equal to the value of the ``task.concurrency``
+configuration parameter (see :ref:`task-properties`).
+
+When the build table is partitioned, the spill-to-disk mechanism can decrease
+the peak memory usage needed by the join operation. When a query approaches the
+memory limit, a subset of the partitions of the build table gets spilled to disk,
+along with rows from the other table that fall into those same partitions. The
+number of partitions that get spilled influence the amount of disk space needed.
```

Comment:
Collective nouns take singular verbs. 
"get" is right ,though.  I guess because the subject of the verb is "partitions" and not "number".  

On the other hand it should should, be 'influences'  rather than 'influence"

so the "number of partitions that get spilled influences the amount of diskspace needed"